subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprint.kt

```diff
@@ -29,7 +29,12 @@ sealed class ConfigurationCacheFingerprint {
     data class GradleEnvironment(
         val gradleUserHomeDir: File,
         val jvm: String,
-        val startParameterProperties: Map<String, Any?>
+        val startParameterProperties: Map<String, Any?>,
+        /**
+         * Whether the instrumentation agent was used when computing the cache.
+         * With the agent, the class paths may be stored differently, making the caches incompatible with one another.
+         */
+        val instrumentationAgentUsed: Boolean
```

Comment:
Perhaps it should be part of the cache key to maximize cache hits as we play with both modes?